x-pack/plugins/security/server/lib/watch_status_and_license_to_initialize.js

```diff
@@ -4,7 +4,24 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import * as Rx from 'rxjs';
-import { map, switchMap, tap } from 'rxjs/operators';
+import { catchError, mergeMap, map, retryWhen, switchMap, tap } from 'rxjs/operators';
+
+export const retryStrategy = ({
+  maxAttempts,
+  scalingDuration,
+}) => (errors) => {
+  return errors.pipe(
+    mergeMap((error, i) => {
+      const attempt = i + 1;
+
+      if (attempt >= maxAttempts) {
+        return Rx.throwError(error);
+      }
+
+      return Rx.timer(attempt * scalingDuration);
+    })
+  );
+};
 
 export function watchStatusAndLicenseToInitialize(xpackMainPlugin, downstreamPlugin, initialize) {
```

Comment:
Now that this could, in theory, be in a retry loop for up to ~21 seconds, what do you think about adding a couple log entries before and after this call, just to give an indication of what's going on?

Reply:
I added the logs to when we're registering the privileges, as this is where all of the logging is currently taking place, I figured those errors being logged plus the yellow status would be sufficient.